storage/dataflux/fast_list.go

```diff
@@ -73,18 +86,68 @@ type Lister struct {
 type CloseFunc func()
 
 // NewLister creates a new dataflux Lister to list objects in the give bucket.
-func NewLister(opts *Input) (*Lister, CloseFunc) {
-
-	lister := &Lister{}
+func NewLister(c *storage.Client, in *ListerInput) (*Lister, CloseFunc) {
+	bucket := c.Bucket(in.BucketName)
+	lister := &Lister{
+		method:               open,
+		pageToken:            "",
+		bucket:               bucket,
+		batchSize:            in.BatchSize,
+		query:                in.Query,
+		skipDirectoryObjects: in.SkipDirectoryObjects,
+	}
 	return lister, func() { lister.Close() }
 }
 
 // NextBatch runs worksteal algorithm and sequential listing in parallel to quickly
-// return a list of objects in the bucket. For buckets with smaller dataset,
-// sequential listing is expected to be faster. For buckets with larger dataset,
+// return a list of objects in the bucket. For smaller dataset,
+// sequential listing is expected to be faster. For larger dataset,
 // worksteal listing is expected to be faster.
 func (c *Lister) NextBatch(ctx context.Context) ([]*storage.ObjectAttrs, error) {
-	return nil, nil
+	// countError tracks the number of failed listing methods.
+	countError := 0
+	var results []*storage.ObjectAttrs
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	g, childCtx := errgroup.WithContext(ctx)
+
+	// Run worksteal listing when method is Open or WorkSteal.
+	// Run sequential listing when method is Open or Sequential.
```

Comment:
So the idea is that method is open to start, and then will be either sequential or workSteal subsequently after the first NextBatch call? If so maybe note that and add TODOs.

Reply:
Done. 
Added explanation and TODO.